src/services/archive.ts

```diff
@@ -0,0 +1,252 @@
+            classes.forEach((newClass, index) => {
+                classes[index].academicYear = academicYear
+            })
+
+            await this.schoolClassArchiveRepository.save(classes)
+        }
+
+    }
+
+    public async moveTeacher(academicYearId: number) {
+
+        const archiveTeachers = await this.teacherRepository.find({ relations: ['subject'] })
+
+        if (archiveTeachers.length <= 0) {
+            throw new Error("No teacher Found");
+        } else {
+            const academicYear = await this.academicYear(academicYearId)
+
+            let teachers = await this.teacherArchiveRepository.save(archiveTeachers)
+
+            teachers.forEach((teacher, index) => {
+                teachers[index].academicYear = academicYear
+            })
+
+            await this.teacherArchiveRepository.save(teachers)
+        }
+
+    }
+
+    public async moveSubject(academicYearId: number) {
+
+        const archiveSubjects = await this.subjectRepository.find({ relations: ['teacher','student'] })
+
+        if (archiveSubjects.length <= 0) {
+            throw new Error("No Subjects Found");
+        } else {
+            const academicYear = await this.academicYear(academicYearId)
+
+            let subjects = await this.subjectArchiveRepository.save(archiveSubjects)
+
+            subjects.forEach((subject, index) => {
+                subjects[index].academicYear = academicYear
+            })
+
+            await this.subjectArchiveRepository.save(subjects)
+        }
+    }
+
+    public async moveExaminations(academicYearId: number) {
+
+        const exams = await this.examinationsRepository.find({ relations: ['teacher'] })
+
+        if (exams.length <= 0) {
+            throw new Error("No Exams Found");
+        } else {
+            const academicYear = await this.academicYear(academicYearId)
+
+            let archiveExams = await this.examinationArchiveRepository.save(exams);
+
+            archiveExams.forEach((exam, index) => {
+                archiveExams[index].academicYear = academicYear
+            })
+
+            const exam = await this.examinationArchiveRepository.save(archiveExams)
+        }
+    }
+
+    public async moveResults(academicYearId: number) {
+
+        const results = await this.resultRepository.find({ relations: ['exam', 'student'] })
+
+        if (results.length <= 0) {
+            throw new Error("No results found");
+        } else {
+            const academicYear = await this.academicYear(academicYearId)
+
+            let archiveResults = await this.resultArchiveRepository.save(results)
+
+            archiveResults.forEach((result, index) => {
+                archiveResults[index].academicYear = academicYear
+            })
+
+            const result = await this.resultArchiveRepository.save(archiveResults)
+        }
+
+    }
+
+    public async removeNormalTables() {
+        const classes = await this.schoolClassRepository.find()
```

Comment:
this is making two database calls, you can remove by passing the criteria to remove using the options, no need to first get them and then delete them.